test/e2e/integration_test.go

```diff
@@ -0,0 +1,59 @@
+package e2e
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	capi_e2e "sigs.k8s.io/cluster-api/test/e2e"
+)
+
+var _ = Describe("When testing integration [integration]", func() {
+	It("CI Test Provision", func() {
+		ciTestProvision()
+	})
+})
+
+func ciTestProvision() {
+	k8sVersion := e2eConfig.GetVariable("KUBERNETES_VERSION")
```

Comment:
please not that by default e2e uses 4 nodes and intergration tests needs only 2 nodes 
https://github.com/metal3-io/cluster-api-provider-metal3/blob/87f2c8466fd444f6ac5951fcdeed80be5031cbed/scripts/ci-e2e.sh#L27
and it should try to provision only 1 kcp and 1 md
https://github.com/metal3-io/cluster-api-provider-metal3/blob/87f2c8466fd444f6ac5951fcdeed80be5031cbed/test/e2e/config/e2e_conf.yaml#L115-L116

Reply:
and I do not think we need this function `ciTestProvision` you can just move `createTargetCluster` to the It block